Shelter-England/contentful-migration-generator
# Script to import Accessplanit courses

This script fetches course data from GraphQL and creates Contentful entries for each course. It creates a page entry and a data entry and then links them together.

I have run this on the "neil-course-import" Contentful env and I can see 178 courses have been created. This a few short of the number I was expecting but I think the few existing Page - Course entries Neil added may account for this. 

There's quite a lot of code here and not a lot of comments. I'll add some explainers next week.

Testing: 
1. Run the `website-assets` repo at `locahost:8000` with `yarn develop` (branch: feature/accessplanit-main)
2. Run this script: from within the courses folder `node createCourseEntries.js`

Running the script right now shouldn't do a lot as it shouldn't create duplicate entries in Contentful. You could delete all Page - Course and Data - Course and test it that way.


There are one or two new changes on the `feature/accessplanit-main` branch of website-assets that impact on this script. The slug/templateIdString fields are now using underscores rather than dashes ("Writing Homelessness Decision Letters" becomes "writing_homelessness_decision_letters" rather than "writing-homelessness-decision-letters"), this felt like the better choice for consistency as using the same value for the `slug` and `templateIdString` makes it easier to link the two entry types in the script. 

~~I can also see that colons haven't been correctly stripped from some course names, I'll fix that on the  `feature/accessplanit-main` branch on Monday~~ (This has been fixed)
